### PR TITLE
add server-timing example in docs

### PR DESCRIPTION
**Pull request by wetteyve**
- this PR adds server-timings to measure performance of loader-functions
- in the docs is an example how to implement this utility

### CODE REVIEW

File: app/utils/server/timing.server.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  string,
+  Array<
+    { desc?: string } & (
+      | { time: number; start?: never }
+      | { time?: never; start: number }
+    )
+  >
+>;
+
+export function makeTimings(type: string, desc?: string) {
```
Comment (illiecemessaoudi) [inline app/utils/server/timing.server.ts:11]: IMO we should have a really Typed "type" that can be helpful for interacting or sorting the timings more precisely

File: app/utils/server/timing.server.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+export async function time<ReturnType>(
+  fn: Promise<ReturnType> | (() => ReturnType | Promise<ReturnType>),
+  {
+    type,
+    desc,
+    timings,
+  }: {
+    type: string;
+    desc?: string;
```
Comment (illiecemessaoudi) [inline app/utils/server/timing.server.ts:46]: this "desc" is for  description ?

File: app/utils/server/timing.server.ts (hunk 1 above)
Comment (wetteyve) [reply app/utils/server/timing.server.ts:11]: when sorting, the timings in the brwoser are unfortunately sorted alphabetically. i added the index with a leading-0 when generating the keys to maintain the order of execution in the code. what do you think? in my opinion, this is the only “meaningful” sorting for these timings.
image
image

File: app/utils/server/timing.server.ts (hunk 2 above)
Comment (wetteyve) [reply app/utils/server/timing.server.ts:46]: i copy pasted it from the epic stack in the beginning but removed it now. i felt that is not generating any additional value as it does not show up in the dev console anywhere

Comment (illiecemessaoudi) [reply app/utils/server/timing.server.ts]: it was just some of my askings, but good

File: app/utils/server/timing.server.ts (hunk 1 above)
Comment (illiecemessaoudi) [reply app/utils/server/timing.server.ts]: yes but i was just thinking if we can group timings by specific type predifined by us ex (Function, Download, Process, Loading... ) Things like that, dont know if a convention exist about this

Comment (wetteyve) [reply app/utils/server/timing.server.ts]: i think we can predefine groups also later on when we are experimenting a little bit.

i would also guess that these timing functions are gonna be mostly used for local development and maybe on stage/ test environments to measure some things here and there. so we dont need to think about grouping to much now allready IMO.

Comment (illiecemessaoudi) [reply app/utils/server/timing.server.ts:11]: Ok good for me

**Review by illiecemessaoudi — APPROVED** [review]

**Review by GauthierBosson — APPROVED** [review]

**Pull request merged**